billing-report-agent/src/main/java/com/epam/pipeline/billingreportagent/service/impl/converter/AwsStorageToBillingRequestConverter.java

```diff
@@ -0,0 +1,297 @@
+import com.epam.pipeline.billingreportagent.service.EntityToBillingRequestConverter;
+import com.epam.pipeline.entity.datastorage.AbstractDataStorage;
+import com.epam.pipeline.entity.datastorage.DataStorageType;
+import com.epam.pipeline.entity.user.PipelineUser;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.elasticsearch.action.DocWriteRequest;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.search.SearchHits;
+import org.elasticsearch.search.aggregations.AggregationBuilders;
+import org.elasticsearch.search.aggregations.metrics.sum.ParsedSum;
+import org.elasticsearch.search.aggregations.metrics.sum.SumAggregationBuilder;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.math.RoundingMode;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.YearMonth;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Slf4j
+@SuppressWarnings("checkstyle:MagicNumber")
+public class AwsStorageToBillingRequestConverter implements EntityToBillingRequestConverter<AbstractDataStorage> {
+
+    private static final String STORAGE_SIZE_AGG_NAME = "sizeSumSearch";
+    private static final String SIZE_FIELD = "size";
+    private static final String REGION_FIELD = "storage_region";
+    private static final int BYTES_TO_GB = 1 << 30;
+    private static final int CENTS_IN_DOLLAR = 100;
+    private static final int PRECISION = 5;
+    private static final String ES_FILE_INDEX_PATTERN = "cp-%s-file-%d";
+    private static final String INDEX_PATTERN = "%s-daily-%d-%s";
+    private static final RoundingMode ROUNDING_MODE = RoundingMode.CEILING;
+
+    private final EntityMapper<StorageBillingInfo> mapper;
+    private final ElasticsearchServiceClient elasticsearchService;
+    private final Map<Regions, StoragePricing> storagePriceListGb = new HashMap<>();
+    private BigDecimal defaultPriceGb;
+    private final StorageType storageType;
+
+    public AwsStorageToBillingRequestConverter(final EntityMapper<StorageBillingInfo> mapper,
+                                               final ElasticsearchServiceClient elasticsearchService,
+                                               final String awsStorageServiceName,
+                                               final StorageType storageType) {
+        this.mapper = mapper;
+        this.elasticsearchService = elasticsearchService;
+        this.storageType = storageType;
+        initPrices(awsStorageServiceName);
```

Comment:
Prices shall be refreshed for each synchronisation cycle.

Reply:
Done